tsconfig.json

```diff
@@ -0,0 +1,11 @@
+{
+  "include": ["lib/**/*", "espree.js"],
+  "compilerOptions": {
+    "allowJs": true,
```

Comment:
`"checkJs": true`?

Reply:
If we do that, we'd have to change a lot of espree's code to avoid warnings about `any`s, possible `undefined`s etc.
At that point I think we could just port the project to typescript instead (It'll take a bit of refactoring but shouldn't be impossible)

Reply:
We don't have a plan to convert to typescript. A promising way to do type-checking based on the jsdocs. Imo, it doesn't have to be in the PR - we can explore it later.